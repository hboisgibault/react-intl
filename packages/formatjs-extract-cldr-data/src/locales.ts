/*
 * Copyright 2015, Yahoo Inc.
 * Copyrights licensed under the New BSD License.
 * See the accompanying LICENSE file for terms.
 */
// These are the exceptions to the default algorithm for determining a locale's
// parent locale.
import * as AVAILABLE_LOCALES from 'cldr-core/availableLocales.json';

export function getAllLocales() {
  return AVAILABLE_LOCALES.availableLocales.full.sort();
}

export function getAllLanguages() {
  return new Set(
    AVAILABLE_LOCALES.availableLocales.full.map(l => l.split('-')[0])
  );
}
