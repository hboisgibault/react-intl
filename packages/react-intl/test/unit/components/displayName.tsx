import * as React from 'react';
import {mount} from 'enzyme';
import {FormattedDisplayName} from '../../../src';
import {mountFormattedComponentWithProvider} from '../testUtils';

const mountWithProvider = mountFormattedComponentWithProvider(
  FormattedDisplayName
);

const intlConfig = {locale: 'en'};

describe('<FormattedDisplayName />', () => {
  it('has a `displayName`', () => {
    expect(FormattedDisplayName.displayName).toBe('FormattedDisplayName');
  });

  it('throws when <IntlProvider> is missing from ancestry', () => {
    const mockConsole = jest
      .spyOn(console, 'error')
      .mockImplementation(() => null);
    try {
      expect(() => mount(<FormattedDisplayName value="zh-Hans" />)).toThrow(
        '[React Intl] Could not find required `intl` object. <IntlProvider> needs to exist in the component ancestry.'
      );
    } finally {
      mockConsole.mockRestore();
    }
  });

  it('accepts Intl.DisplayNames options', () => {
    const rendered = mountWithProvider(
      {
        type: 'currency',
        value: 'CNY',
      },
      intlConfig
    );
    expect(rendered).toMatchSnapshot();
  });

  it('renders an empty <> when the underlying DisplayNames would return undefined', () => {
    // When fallback is none, it will return undefined if no display name is available.
    const displayNames = new (Intl as any).DisplayNames('en', {
      fallback: 'none',
    });
    expect(displayNames.of('xx-XX')).toBeUndefined();

    // Now let's do the same with <FormattedDisplayNames />
    const rendered = mountWithProvider(
      {
        fallback: 'none',
        value: 'xx-XX',
      },
      intlConfig
    );
    expect(rendered.text()).toBe('');
    expect(rendered).toMatchSnapshot();
  });
});
