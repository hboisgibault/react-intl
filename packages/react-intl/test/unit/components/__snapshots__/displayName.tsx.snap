// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormattedDisplayName /> accepts Intl.DisplayNames options 1`] = `
<FormattedDisplayName
  type="currency"
  value="CNY"
>
  Chinese Yuan
</FormattedDisplayName>
`;

exports[`<FormattedDisplayName /> renders an empty <> when the underlying DisplayNames would return undefined 1`] = `
<FormattedDisplayName
  fallback="none"
  value="xx-XX"
/>
`;
